File: crates/infergen-core/templates/flask/snippet.py
# After `infergen generate`, import your typed SDK:
from infergen_generated import track

@app.before_request
def track_request():
    track.route_accessed(endpoint=request.endpoint or "", method=request.method)

@app.route("/login", methods=["POST"])
def login():
    user = authenticate(request.form)
    if user:
        track.user_logged_in(user_id=str(user.id))
    return redirect("/dashboard")

File: crates/infergen-core/templates/gin/snippet.go
// After `infergen generate`, import your typed SDK:
import infergen "github.com/your-org/your-project/infergen/generated"

func TrackingMiddleware() gin.HandlerFunc {
	return func(c *gin.Context) {
		infergen.Track.HttpRequestHandled(c.Request.Method, c.FullPath())
		c.Next()
	}
}

func LoginHandler(c *gin.Context) {
	userID := authenticate(c)
	infergen.Track.UserAuthenticated(userID)
	c.JSON(200, gin.H{"ok": true})
}
